actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -403,10 +403,18 @@ def logger
     def commit_flash
     end
 
+    def ssl?
+      super || secure_websocket_request?
+    end
+
     private
       def check_method(name)
         HTTP_METHOD_LOOKUP[name] || raise(ActionController::UnknownHttpMethod, "#{name}, accepted HTTP methods are #{HTTP_METHODS[0...-1].join(', ')}, and #{HTTP_METHODS[-1]}")
         name
       end
+
+      def secure_websocket_request?
+        get_header('Upgrade') && get_header('Upgrade').downcase == 'websocket' && get_header('rack.url_scheme') == 'wss'
```

Comment:
Isn't it strictly true that `wss` is secure (i.e., `ssl? == true`), regardless of whether we're doing an upgrade?